### PR TITLE
feat: list third party collections

### LINKED ISSUE #1630: List Third Party Collections
We need to show all the third party collections created in the builder a wallet is a manager of

Don't forget the subtitle to differentiate from DCL collections (https://github.com/decentraland/builder/issues/1562)

### LINKED ISSUE #1650: Update the urn module so it exports decodes for each kind of URN
Instead of exporting `join` and `pop`, export methods like:

decodeOffChainURN
decodeCollectionURN
etc

which will return an object with each part of the URN already available to use

### PR DESCRIPTION
**Pull request by nicosantangelo**
image

Missing tests for the newly introduced decodeURN method

Closes #1630 
Closes #1650

### CODE REVIEW

File: src/lib/urn.ts [hunk 1]
```diff
@@ -1,25 +1,106 @@
...
+ *   )
+ */
+const baseMatcher = 'urn:decentraland'
+const protocolMatcher = '(?<protocol>mainnet|ropsten|matic|mumbai|off-chain)'
+const typeMatcher = '(?<type>base-avatars|collections-v2|collections-thirdparty)'
+
+const baseAvatarsSuffixMatcher = '((?<=base-avatars:)BaseMale|BaseFemale)'
+const collectionsSuffixMatcher = '((?<=collections-v2:)0x[a-fA-F0-9]{40})'
+const thirdPartySuffixMatcher =
+  '((?<=collections-thirdparty:)(?<thirdPartyName>[^:|\\s]+)(:(?<thirdPartyCollectionId>[^:|\\s]+))?(:(?<thirdPartyTokenId>[^:|\\s]+))?)'
```
Comment (LautaroPetaccio) [inline src/lib/urn.ts:39]: Regex capture groups are around 3 year old and they don't have polyfills, do you know by any change which browser versions should we support?

File: src/lib/urn.spec.ts [hunk 2]
```diff
@@ -15,28 +15,126 @@ describe('when building the item URN', () => {
...
+    it('should decode and return each group', () => {
+      expect(decodeURN('urn:decentraland:ropsten:collections-v2:0xc6d2000a7a1ddca92941f4e2b41360fe4ee2abd8')).toEqual({
+        type: URNType.COLLECTIONS_V2,
+        protocol: URNProtocol.ROPSTEN,
+        suffix: '0xc6d2000a7a1ddca92941f4e2b41360fe4ee2abd8'
+      })
+    })
+  })
+
+  describe('when a valid third party', () => {
```
Comment (LautaroPetaccio) [inline src/lib/urn.spec.ts:98]:
```suggestion
  describe('when decoding a valid third party', () => {
```

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

File: src/modules/collection/utils.ts [hunk 3]
```diff
@@ -95,3 +96,11 @@ export function getMostRelevantStatus(statusA: SyncStatus, statusB: SyncStatus)
...
   return indexA < indexB ? statusA : statusB
 }
+
+export function isThirdParty(collection: Collection) {
+  if (!collection.urn) {
+    return false
+  }
+  const decodedURN = decodeURN(collection.urn)
+  return decodedURN.type === URNType.COLLECTIONS_THIRDPARTY
+}
```
[the remark this answers is not in the document]
Comment (nicosantangelo) [reply src/modules/collection/utils.ts:106]: Completely forgot I added this. Yes!

File: src/lib/urn.ts (hunk 1 above)
Comment (nicosantangelo) [reply src/lib/urn.ts:39]: Checked when developing and found this:
image

https://caniuse.com/?search=regex%20capture

2020 for edge and FF, but saw `Global 93.2%` and went ahead with it

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**